thorlcr/slave/slwatchdog.cpp

```diff
@@ -132,6 +132,35 @@ class CGraphProgressHandlerBase : public CSimpleInterface, implements ISlaveWatc
             activeGraphs.zap(graph);
         }
     }
+    virtual void debugRequest(CMessageBuffer &msg, const char *request) const
+    {
+        Owned<IPTree> req = createPTreeFromXMLString(request);
+
+        StringBuffer edgeString, sActId;
+        req->getProp("@edgeId", edgeString);
+        // Make sure it has sensible values
+        if (!edgeString.length() || edgeString.length()>99 ) return;
```

Comment:
>99 a little arbitrary
I supposed strictly speaking limit is max digit length of 2 unsigned 32bit numbers + '_' 

But I'd probably just validate that when you perform the atoi's and make sure actId is not 0 too.